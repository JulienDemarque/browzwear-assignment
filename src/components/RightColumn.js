import React, { Component } from "react";

class RightColumn extends Component {
  render() {
    const { companies, selectedCompany, handleSelectCompany } = this.props;

    return (
      <div className="table-container__column">
        <h3 className="column__title">Company</h3>
        <ul>
          {companies.map(({ companyName }) => (
            <li key={companyName}>
              <button
                className={
                  companyName === selectedCompany ? "button_highlight" : ""
                }
                onClick={handleSelectCompany}
                value={companyName}
              >
                <span>{companyName}</span>
              </button>
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

export default RightColumn;
