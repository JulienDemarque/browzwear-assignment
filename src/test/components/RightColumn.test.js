import React from "react";
import { shallow } from "enzyme";

import RightColumn from "../../components/RightColumn";
import { RightColumnProps } from "../data-for-testing";
const { companies, selectedCompany } = RightColumnProps;

describe("RightColumn Component", () => {
  let wrapper;
  const mockHandleSelectCompany = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <RightColumn
        companies={companies}
        selectedCompany={selectedCompany}
        handleSelectCompany={mockHandleSelectCompany}
      />
    );
  });

  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
