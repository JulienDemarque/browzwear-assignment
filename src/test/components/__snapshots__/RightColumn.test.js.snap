// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightColumn Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RightColumn
    companies={
      Array [
        Object {
          "companyName": "Lonesome Pine Restaurant",
          "fromCity": "Portland",
        },
        Object {
          "companyName": "The Big Cheese",
          "fromCity": "Portland",
        },
      ]
    }
    handleSelectCompany={[MockFunction]}
    selectedCompany="Lonesome Pine Restaurant"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="column__title"
        >
          Company
        </h3>,
        <ul>
          <li>
            <button
              className="button_highlight"
              onClick={[MockFunction]}
              value="Lonesome Pine Restaurant"
            >
              <span>
                Lonesome Pine Restaurant
              </span>
            </button>
          </li>
          <li>
            <button
              className=""
              onClick={[MockFunction]}
              value="The Big Cheese"
            >
              <span>
                The Big Cheese
              </span>
            </button>
          </li>
        </ul>,
      ],
      "className": "table-container__column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Company",
          "className": "column__title",
        },
        "ref": null,
        "rendered": "Company",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <button
                className="button_highlight"
                onClick={[MockFunction]}
                value="Lonesome Pine Restaurant"
              >
                <span>
                  Lonesome Pine Restaurant
                </span>
              </button>
            </li>,
            <li>
              <button
                className=""
                onClick={[MockFunction]}
                value="The Big Cheese"
              >
                <span>
                  The Big Cheese
                </span>
              </button>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Lonesome Pine Restaurant",
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button_highlight"
                onClick={[MockFunction]}
                value="Lonesome Pine Restaurant"
              >
                <span>
                  Lonesome Pine Restaurant
                </span>
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Lonesome Pine Restaurant
                </span>,
                "className": "button_highlight",
                "onClick": [MockFunction],
                "value": "Lonesome Pine Restaurant",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lonesome Pine Restaurant",
                },
                "ref": null,
                "rendered": "Lonesome Pine Restaurant",
                "type": "span",
              },
              "type": "button",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "The Big Cheese",
            "nodeType": "host",
            "props": Object {
              "children": <button
                className=""
                onClick={[MockFunction]}
                value="The Big Cheese"
              >
                <span>
                  The Big Cheese
                </span>
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  The Big Cheese
                </span>,
                "className": "",
                "onClick": [MockFunction],
                "value": "The Big Cheese",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "The Big Cheese",
                },
                "ref": null,
                "rendered": "The Big Cheese",
                "type": "span",
              },
              "type": "button",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="column__title"
          >
            Company
          </h3>,
          <ul>
            <li>
              <button
                className="button_highlight"
                onClick={[MockFunction]}
                value="Lonesome Pine Restaurant"
              >
                <span>
                  Lonesome Pine Restaurant
                </span>
              </button>
            </li>
            <li>
              <button
                className=""
                onClick={[MockFunction]}
                value="The Big Cheese"
              >
                <span>
                  The Big Cheese
                </span>
              </button>
            </li>
          </ul>,
        ],
        "className": "table-container__column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Company",
            "className": "column__title",
          },
          "ref": null,
          "rendered": "Company",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <button
                  className="button_highlight"
                  onClick={[MockFunction]}
                  value="Lonesome Pine Restaurant"
                >
                  <span>
                    Lonesome Pine Restaurant
                  </span>
                </button>
              </li>,
              <li>
                <button
                  className=""
                  onClick={[MockFunction]}
                  value="The Big Cheese"
                >
                  <span>
                    The Big Cheese
                  </span>
                </button>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Lonesome Pine Restaurant",
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="button_highlight"
                  onClick={[MockFunction]}
                  value="Lonesome Pine Restaurant"
                >
                  <span>
                    Lonesome Pine Restaurant
                  </span>
                </button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Lonesome Pine Restaurant
                  </span>,
                  "className": "button_highlight",
                  "onClick": [MockFunction],
                  "value": "Lonesome Pine Restaurant",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lonesome Pine Restaurant",
                  },
                  "ref": null,
                  "rendered": "Lonesome Pine Restaurant",
                  "type": "span",
                },
                "type": "button",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "The Big Cheese",
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className=""
                  onClick={[MockFunction]}
                  value="The Big Cheese"
                >
                  <span>
                    The Big Cheese
                  </span>
                </button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    The Big Cheese
                  </span>,
                  "className": "",
                  "onClick": [MockFunction],
                  "value": "The Big Cheese",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "The Big Cheese",
                  },
                  "ref": null,
                  "rendered": "The Big Cheese",
                  "type": "span",
                },
                "type": "button",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
